onebusaway-android/src/main/java/org/onebusaway/android/ui/TripPlanFragment.java

```diff
@@ -0,0 +1,464 @@
+                    break;
+                }
+            }
+        }
+
+        final boolean unitsAreImperial = !RegionUtils.getUnitsAreMetric(getContext());
+
+        Double maxWalk = builder.getMaxWalkDistance();
+        if (maxWalk != null && Double.MAX_VALUE != maxWalk) {
+            if (unitsAreImperial)
+                maxWalk *= 3.281;
+            maxWalkEditText.setText(String.format("%d", maxWalk.longValue()));
+        }
+
+
+        if (unitsAreImperial) {
+            TextView label = (TextView) dialog.findViewById(R.id.label_minimum_walk_distance);
+            label.setText(getString(R.string.distance_feet));
+        }
+
+        dialog.findViewById(R.id.advanced_settings_ok).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                dialog.dismiss();
+
+                boolean optimizeTransfers = minimizeTransfersCheckbox.isChecked();
+
+                int modeId = transitModeResource.getResourceId(spinnerTravelBy.getSelectedItemPosition(), 0);
+
+                boolean wheelchair = wheelchairCheckbox.isChecked();
+
+                String maxWalkString = maxWalkEditText.getText().toString();
+                double maxWalkDistance;
+                if (TextUtils.isEmpty(maxWalkString)) {
+                    maxWalkDistance = Double.MAX_VALUE;
+                } else {
+                    double d = Double.parseDouble(maxWalkString);
+                    maxWalkDistance = unitsAreImperial ? d / 3.281 : d;
+                }
+
+                builder.setOptimizeTransfers(optimizeTransfers)
+                        .setModeSetById(modeId)
+                        .setWheelchairAccessible(wheelchair)
+                        .setMaxWalkDistance(maxWalkDistance);
+            }
+        });
+    }
+
+    private void makeNoLocationToast() {
+        Toast.makeText(getContext(), getString(R.string.tripplanner_error_no_location), Toast.LENGTH_SHORT).show();
+    }
+
+    private CustomAddress makeAddressFromLocation() {
+        Locale locale = Locale.getDefault();
+
+        Location loc = Application.getLastKnownLocation(getContext(), mGoogleApiClient);
+        if (loc == null) {
+            if (getContext() != null)
+                Toast.makeText(getContext(), getString(R.string.main_location_unavailable), Toast.LENGTH_SHORT).show();
+            return null;
+        }
+        CustomAddress address = new CustomAddress(locale);
+        address.setAddressLine(0, getString(R.string.tripplanner_current_location));
+        address.setLatitude(loc.getLatitude());
+        address.setLongitude(loc.getLongitude());
+        return address;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        if (resultCode != -1) {
+            Log.e(TAG, "Error getting geocoding results");
+            return;
+        }
+
+        Place place = PlaceAutocomplete.getPlace(getActivity(), intent);
```

Comment:
Per https://developers.google.com/android/reference/com/google/android/gms/location/places/ui/PlaceAutocomplete.html#public-methods, this should use the Application Context, not the activity context:

`Place place = PlaceAutocomplete.getPlace(Application.get().getApplicationContext(), intent);`